notification/src/interfaces/sandboxExecutor.ts

```diff
@@ -1,3 +1,3 @@
 export default interface SandboxExecutor {
-  evaluate(expression: string, data: object): boolean;
+  evaluate(expression: string, data: object | undefined): boolean;
```

Comment:
Why also accept undefined? Is there a design decision to make?

Reply:
Yes. The data ofJobresult of the tranformation (sandbox execution) may return undefined on failure. I imagined the scenario that the user wants only to be informed if the pipeline/transformation fails. He can set the condition to "data == undefined" for that purpose.

I may consider changing the data to null on failure in the (transformation-)sandbox executor, so the user can check for the condition "data ==null".
(Does this make a difference at all?)

Reply:
Sounds like a good decision! Make sure to document it somewhere

Reply:
Makes sense! Make sure to document it somewhere ;)